### PR TITLE
Correcting few corner case in validation evaluation during training

### PR DESCRIPTION
**Pull request by vladgets**

### CODE REVIEW

File: nemo/collections/nlp/data/question_answering_squad/qa_dataset.py [hunk 1]
```diff
@@ -136,6 +130,14 @@ def __init__(
                     with open(cached_features_file, "wb") as writer:
                         pickle.dump(self.features, writer)
 
+        # check number of samples. Should be either -1 not to limit or positive number
+        if num_samples == 0:
+            raise ValueError(
```
Comment (yzhang123) [inline nemo/collections/nlp/data/question_answering_squad/qa_dataset.py:135]: Could you move this section up before line 110, then you can remove index checks down below

**Review by yzhang123 — CHANGES_REQUESTED** [review]

File: nemo/collections/nlp/data/question_answering_squad/qa_dataset.py (hunk 1 above)
Comment (tkornuta-nvidia) [reply nemo/collections/nlp/data/question_answering_squad/qa_dataset.py:135]: wait... you mean: before use_cache?

Comment (yzhang123) [reply nemo/collections/nlp/data/question_answering_squad/qa_dataset.py:135]: I think its best to move up line 134- 139 before line 110

**Review by yzhang123 — APPROVED** [review]
Thank you @vladgets !

File: nemo/collections/nlp/data/question_answering_squad/qa_dataset.py (hunk 1 above)
Comment (vladgets) [reply nemo/collections/nlp/data/question_answering_squad/qa_dataset.py]: Corrected

**Pull request merged**